uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -341,24 +354,32 @@ define([
        * Also, checks if its an update request from manager and accordingly updates leave request
        */
       this.submit = function () {
-        if (this.isMode('view') || this.submitting) {
-          return;
-        }
+        var originalStatus, promise;
 
-        // current absence type (this.request.type_id) doesn't allow self
-        if (this.balance.closing < 0 && this.selectedAbsenceType.allow_overuse === '0') {
-          // show an error
-          this.errors = ['You are not allowed to apply leave in negative'];
+        if(validateBeforeSubmit.call(this)) {
           return;
         }
 
         this.errors = [];
 
+        originalStatus = this.request.status_id;
+        changeStatusBeforeSave.call(this);
+
+        this.submitting = true;
         if (canViewOrEdit.call(this)) {
```

Comment:
True its not and should be analyzed, but I do not want to spend more time on this ticket as the UI is working fine. Adding a TODO instead.